README.md

```diff
@@ -75,6 +77,43 @@ generic with minimal need for project-specific behavior.
 [Docker]: https://docs.docker.com/install/overview/
 
 
+### 2.1.1 git-secrets
+
+[git-secrets] helps prevent secrets (passwords, credentials, etc.) from being
+submitted into a Git repository. See the *Installing git-secrets* section of the
+project's README for instructions how to install [git-secrets] on your OS.
+
+Once installed, git-secrets will need to be configured for your local repository
+clones. To configure git-secrets in a single clone, run:
+
+    cd /path/to/my/clone
+    git secrets --install
+    git secrets --register-aws
+    
+To automatically configure git-secrets in all future clones, run:
+
+    git secrets --install ~/.git-templates/git-secrets
+    git config --global init.templateDir ~/.git-templates/git-secrets
+    git secrets --register-aws --global
+
+Finally, once configured you should verify the functionality of git-secrets. To
+do this, create a text file with the contents below. Remove all occurrences of
+`[x]` from the text file and attempt to commit the file to your each of your
+clones.
+
+git-secrets test file:
+
+    AKIA000000000000000[x]0
+    AWS_SECRET_ACCESS_KEY=000000000000000000000000000000000000000[x]0
+    AWS_ACCOUNT_ID=0000-0000-000[x]0
+
+If git-secrets is configured correctly, the commit will fail to complete. If
+the commit succeeds, or if the error message does not mention all three lines
+containing secrets, git-secrets is not functioning correctly in your clone.
```

Comment:
Please simplify this. We don't gain much by testing three patterns instead of just one. Convert the instructions to a single sequence of commands using `cat` and `git commit`. See if `git commit` has a dry-run flag so that the hooks are invoked but no commit is made.  In the cat invocation you could use shell quoting to achieve the same as manually doing the `x` thing.

Reply:
Unfortunately, `--dry-run` does not invoke the hooks. In the `man githooks` documentation there are options to bypass the various hooks, but I can find no option to include a hook in a dry run commit.

```
(.venv) daniel@Crispin ~/repo/azul1 $ git s --short
A  git_secrets_test_file.txt

(.venv) daniel@Crispin ~/repo/azul1 $ git commit -m 'foo' --dry-run
On branch issues/danielsotirhos/4858-add-git-secrets-to-prerequisites
Your branch is up to date with 'github/issues/danielsotirhos/4858-add-git-secrets-to-prerequisites'.

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	new file:   git_secrets_test_file.txt

(.venv) daniel@Crispin ~/repo/azul1 $ git commit -m 'foo'
git_secrets_test_file.txt:1:AWS_ACCOUNT_ID=0000-0000-0000

[ERROR] Matched one or more prohibited patterns

Possible mitigations:
- Mark false positives as allowed using: git config --add secrets.allowed ...
- Mark false positives as allowed by adding regular expressions to .gitallowed at repository's root directory
- List your configured patterns: git config --get-all secrets.patterns
- List your configured allowed patterns: git config --get-all secrets.allowed
- List your configured allowed patterns in .gitallowed at repository's root directory
- Use --no-verify if this is a one-time false positive
```

Reply:
Try `git hook run pre-commit`